src/main/java/ch/fhnw/algd1/converters/utf8/UTF8Converter.java

```diff
@@ -3,22 +3,66 @@
 /*
  * Created on 05.09.2014
  */
+
 /**
- * @author 
+ * @author
  */
 public class UTF8Converter {
 	public static byte[] codePointToUTF(int x) {
-		byte[] b = null;
-		// TODO allocate b in the right size (depending on x) and fill it with the
+		byte[] b;
+
+		if (x < 0x80) {
+			b = new byte[1];
+			b[0] = (byte) x;
+			return b;
+		}
+
+		int byteCount = 1;
+		for (int i = 0; i < 4; i++) {
+			if ((x-127) >>> (i*8) != 0) {
+				byteCount++;
+			}
+		}
+
+		b = new byte[byteCount];
+
+		if (byteCount == 2) {
+			b[0] = (byte) (0xC0 | ((x >>> 6) & 0x1F));
```

Comment:
Hier liesse sich kürzen, denn wenn x nur 11 Bits benötigt, sind alle hier "wegmaskierten" Bits ohnehin 0...
```suggestion
			b[0] = (byte) (0xC0 | (x >>> 6));
```